app/services/asset_manager_service.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "gds_api/asset_manager"
+
+class AssetManagerService
+  def upload(file)
+    upload = asset_manager.create_asset(file: file, draft: true)
+    upload["file_url"]
+  end
+
+  def publish(asset)
+    id = asset_id(asset.url)
+    asset_manager.update_asset(id, file: asset, draft: false)
+  end
+
+private
+
+  def asset_manager
+    @asset_manager ||= GdsApi::AssetManager.new(Plek.new.find("asset-manager"))
+  end
+
+  def asset_id(file_url)
+    url_array = file_url.split("/")
+    url_array[url_array.length - 2]
```

Comment:
It should return the 2nd to last element - perhaps `url_array[-2]` is clearer?

Reply:
Yeah, I understand technically what it's doing, but I've no idea why. What does it achieve? Why is the number '2' so important here? (I did actually go and work out why, but it would be helpful to make it clearer here.)